src/Rmath.jl

```diff
@@ -42,14 +42,26 @@ end
     export dwilcox,pwilcox,qwilcox,rwilcox # Wilcox's Rank Sum statistic (m, n)
     export ptukey, qtukey              # Studentized Range Distribution - p and q only
 
-function __init__()
-    # initialize RNG hooks
-    unsafe_store!(cglobal((:unif_rand_ptr,libRmath),Ptr{Cvoid}),
-                  cfunction(rand,Float64,Tuple{}))
-    unsafe_store!(cglobal((:norm_rand_ptr,libRmath),Ptr{Cvoid}),
-                  cfunction(randn,Float64,Tuple{}))
-    unsafe_store!(cglobal((:exp_rand_ptr,libRmath),Ptr{Cvoid}),
-                  cfunction(Random.randexp,Float64,Tuple{}))
+if VERSION < v"0.7.0-DEV.4749"
+    function __init__()
+        # initialize RNG hooks
+        unsafe_store!(cglobal((:unif_rand_ptr,libRmath),Ptr{Cvoid}),
+                      cfunction(rand,Float64,Tuple{}))
+        unsafe_store!(cglobal((:norm_rand_ptr,libRmath),Ptr{Cvoid}),
+                      cfunction(randn,Float64,Tuple{}))
+        unsafe_store!(cglobal((:exp_rand_ptr,libRmath),Ptr{Cvoid}),
+                      cfunction(Random.randexp,Float64,Tuple{}))
+    end
+else
+    function __init__()
+        # initialize RNG hooks
+        unsafe_store!(cglobal((:unif_rand_ptr,libRmath),Ptr{Cvoid}),
+                      @cfunction(rand,Float64,()))
```

Comment:
The macro is being expanded too early which is causing failures on 0.6, so what I think you could do is reverse the conditional and put this stuff first in the first block with a `@static` on the condition.

Reply:
I understand reversing the conditional.  Can you expand a bit on where the `@static` would go?

Reply:
```julia
@static if VERSION >= v"0.7.0-DEV.4749"
    # stuff with @cfunction
else
    # stuff with cfunction
end
```